### PR TITLE
[BEAM-2725] remove nullFields in BeamRecord

### PR DESCRIPTION
**Pull request by XuMingmin**
R: @robertwb

### CODE REVIEW

**Comment by XuMingmin** [comment]
R: + @takidau

File: sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BeamRecordCoder.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public BeamRecordType getRecordType() {
   @Override
   public void encode(BeamRecord value, OutputStream outStream)
       throws CoderException, IOException {
-    nullListCoder.encode(value.getNullFields(), outStream);
+    nullListCoder.encode(scanNullFields(value), outStream);
```
Comment (takidau) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BeamRecordCoder.java]: Are the normal coders unable to encode a null value directly by themselves?

Comment (XuMingmin) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BeamRecordCoder.java]: many `Coder`s cannot encode a null value, like `BigDecimalCoder`, `BigEndianIntegerCoder`, `BigEndianLongCoder`, `StringUtf8Coder` and so on.

Here I handle null values separately, another choice is using NullableCoder to wrap each `Coder`. Is the latter one preferred ?

Comment (takidau) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BeamRecordCoder.java]: No, what you have is probably preferable then. Thanks!

**Comment by takidau** [comment]
Merged, feel free to close.

**Comment by XuMingmin** [comment]
closed #3691, thanks @takidau !

**Pull request merged**